cvu/utils/general.py

```diff
@@ -86,3 +90,50 @@ def unzip_file(filepath: str,
     if clean_up:
         os.remove(filepath)
     return True
+
+
+def read_images_in_batch(img_dir, batchsize, preprocess=None):
+    """
+    Read preprocessed image files in a batch.
+
+    Args:
+        img_dir (str): Path to the directory containing the images.
+
+    Yields:
+        np.ndarray: Batch of preprocessed images.
+    """
+    img_files = [os.path.join(img_dir, f) for f in os.listdir(img_dir)]
+
+    batch = []
+    for img_file in  img_files:
+        if len(batch) == batchsize:
+            yield np.array(batch).squeeze(0)
+            batch = []
+
+        # Read image.
+        image = cv2.imread(img_file)
+
+        # Preprocess
+        if preprocess:
+            image = apply(image, preprocess)
+        batch.append(image)
+    return np.array(batch)
+
+
+def apply(
+        value: np.ndarray,
```

Comment:
minor: make apply more generic for potential usage.  
```suggestion
        value: Any,
```